packages/core/src/modules/message-pickup/services/MessagePickupSessionService.ts

```diff
@@ -34,7 +34,10 @@ export class MessagePickupSessionService {
 
     eventEmitter
       .observable<TransportSessionRemovedEvent>(TransportEventTypes.TransportSessionRemoved)
-      .pipe(takeUntil(stop$))
+      .pipe(
+        filter((e) => e.payload.session.type === 'Websocket'),
```

Comment:
This was actually the key to making it work. The other changes do fix the message delivery not using the websocket but the agent is still sending the status message via http and it's not breaking the pickupRepository.

I noticed in QueueOnly mode it doesn't actually create pickup sessions, so it continues to work correctly, the problem was it wasn't filtering them here so in live mode the http session were removing the pickup session for the websocket. This was what my patch was doing put in a way more complicated way.

I think the status message should also go through `ws` transport but I haven't figured out where that was happening and it's no longer causing the problem.